tests/interpreter/cases/negation/tests.yaml

```
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+cases:
+  - note: not-false
+    data: {}
+    modules:
+      - |
+        package test
+
+        x[a] {
+          not 1 == 2
+          a := "hello"
+        }
+    query: data.test
+    want_result:
+      x: 
+        set!: [hello]
+
+  - note: not-undefined
+    data: {}
+    modules:
+      - |
+        package test
+
+        myequal(t) {
+          t == 2
+        }
+
+        x[a] {
+          not myequal(1)
+          a := "hello"
+        }
+    query: data.test
+    want_result:
+      x: 
+        set!: [hello]
+
+  - note: not-more-undefined
```

```suggestion
  - note: not-more-undefined-OPA-INCOMPATIBLE
```
We could used a naming convention such as above to keep track of behavioral differences from OPA.
Also add the link to https://github.com/open-policy-agent/opa/issues/1877 in comments.

---

Done